### PR TITLE
Pub 1756 - Lefthook

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link
https://tools.hmcts.net/jira/browse/PUB-1756

### Change description
Implemented lefthook with some basic pre-commit options.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by KianKwa — DISMISSED** [review]

**Review by joshblackmoor — APPROVED** [review]

File: lefthook.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            audit:
+                run: yarn audit | cat
+pre-commit:
+    parallel: true
+    commands:
+        eslint:
+            glob: '*.{js,ts,jsx,tsx}'
+            run: yarn eslint --fix {staged_files}
+        prettier:
+            run: yarn prettier --write {staged_files}
```
Comment (ChrisS1512) [inline lefthook.yml:20]: I think we need to add the glob to this prettier step.

I'm hitting an error on the prettier step when making a change to the yarn audit known issues file (No parser found).

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: lefthook.yml (hunk 1 above)
Comment (furnivall) [reply lefthook.yml:21]: done :)

File: package.json [hunk 2]
```diff
@@ -161,5 +162,5 @@
         "ua-parser-js": "1.0.33",
         "http-cache-semantics": "^4.1.1"
     },
-    "packageManager": "yarn@3.3.1"
+    "packageManager": "yarn@3.4.1"
```
Comment (furnivall) [inline package.json:165]: Might be worth mentioning I've snuck in a yarn update without telling anyone...

File: lefthook.yml (hunk 1 above)
Comment (KianKwa) [reply lefthook.yml:21]: Your original prettier PR changed the format in .json, .yml and .md files, etc. Are you sure u want to limit to these 4 files in glob?

Comment (furnivall) [reply lefthook.yml:21]: To be honest, I should switch this! Cheers :)

**Review by KianKwa — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**